src/onig-string-context.h

```diff
@@ -0,0 +1,32 @@
+#ifndef SRC_ONIG_STRING_CONTEXT_H_
+#define SRC_ONIG_STRING_CONTEXT_H_
+
+#include <string>
+#include "nan.h"
+
+using ::v8::Handle;
+using ::v8::Isolate;
+using ::v8::Persistent;
+using ::v8::String;
+using ::v8::Value;
+
+using ::std::string;
+
+class OnigStringContext {
+ public:
+  explicit OnigStringContext(Handle<String> v8String);
+  bool IsSame(Handle<String> other) const;
+  bool HasMultibyteCharacters() const;
+  const char* utf8_value() const { return *utf8Value; }
+  size_t utf8_length() const { return utf8Value.length(); }
+  const wchar_t* utf16_value() const { return reinterpret_cast<const wchar_t*>(*utf16Value); }
```

Comment:
Since `utf16_value` is only used on Windows, I think we can only define it on Windows too, in case we accidentally used `utf16_value` on POSIX in future.